net/net.go

```diff
@@ -240,62 +235,27 @@ func (n *net) CreateThread(_ context.Context, id thread.ID, opts ...core.NewThre
 	return n.getThreadWithAddrs(id)
 }
 
-func (n *net) ensureUnique(id thread.ID, key crypto.Key) error {
-	// Check if thread exists.
-	thrd, err := n.store.GetThread(id)
-	if errors.Is(err, lstore.ErrThreadNotFound) {
-		return nil
-	}
-	if err != nil {
-		return err
-	}
-	// Early out if no log key required.
-	if key == nil {
-		return nil
-	}
-	// Ensure we don't already have this log in our store.
-	if _, ok := key.(crypto.PrivKey); ok {
-		// Ensure we don't already have our 'own' log.
-		if thrd.GetOwnLog().PrivKey != nil {
-			return lstore.ErrThreadExists
-		}
-		return nil
-	}
-	pubKey, ok := key.(crypto.PubKey)
-	if !ok {
-		return fmt.Errorf("invalid log-key")
-	}
-	logID, err := peer.IDFromPublicKey(pubKey)
-	if err != nil {
-		return err
-	}
-	_, err = n.store.GetLog(id, logID)
-	if err == nil {
-		return lstore.ErrLogExists
-	}
-	if !errors.Is(err, lstore.ErrLogNotFound) {
-		return err
-	}
-	return nil
-}
-
 func (n *net) AddThread(ctx context.Context, addr ma.Multiaddr, opts ...core.NewThreadOption) (info thread.Info, err error) {
 	args := &core.NewThreadOptions{}
 	for _, opt := range opts {
 		opt(args)
 	}
-	if _, err = args.Token.Validate(n.getPrivKey()); err != nil {
-		return
-	}
 
 	id, err := thread.FromAddr(addr)
 	if err != nil {
 		return
 	}
-	if err = id.Validate(); err != nil {
+	identity, err := n.Validate(id, args.Token, false)
+	if err != nil {
 		return
 	}
-	if err = n.ensureUnique(id, args.LogKey); err != nil {
+	if identity != nil {
```

Comment:
If an identity is not given via a `thread.Token`, we fall back to using the host's identity.